corehq/apps/userreports/models.py

```diff
@@ -466,24 +466,14 @@ def _get_domain_filter_spec(self):
     def named_expression_objects(self):
         named_expression_specs = deepcopy(self.named_expressions)
         named_expressions = {}
-        spec_error = None
-        factory_context = FactoryContext(named_expressions=named_expressions, named_filters={}, domain=self.domain)
-        while named_expression_specs:
-            number_generated = 0
-            for name, expression in list(named_expression_specs.items()):
-                try:
-                    factory_context.named_expressions.replace(named_expressions)
-                    named_expressions[name] = ExpressionFactory.from_spec(expression, factory_context)
-                    number_generated += 1
-                    del named_expression_specs[name]
-                except BadSpecError as bad_spec_error:
-                    # maybe a nested name resolution issue, try again on the next pass
-                    spec_error = bad_spec_error
-            if number_generated == 0 and named_expression_specs:
-                # we unsuccessfully generated anything on this pass and there are still unresolved
-                # references. we have to fail.
-                assert spec_error is not None
-                raise spec_error
+        factory_context = FactoryContext.empty(self.domain)
+        for name, expression in list(named_expression_specs.items()):
```

Comment:
Why is `list()` needed here?

Reply:
It isn't, just a holdover from the previous code. c00805aec479151cb86bab217a9e139153eb9f75